### ISSUE TITLE
Markdown generator does not consider extension mapping for files with no extension

### ISSUE THREAD

**Issue by qeef** [doxygen/doxygen#7120]
I wanted to generate the documentation for a small project. I have `README.md` and `LICENSE` files in the project root folder.

`README.md` contains:
```
The project is published under [License][1].

[1]: ./LICENSE
```

My `Doxyfile` (also in root) includes:
```
INPUT                  = README.md LICENSE
USE_MDFILE_AS_MAINPAGE = README.md
EXTENSION_MAPPING      = no_extension=md
```

Then I generate documentation with `/usr/local/bin/doxygen` from the project root folder (Doxygen is compiled from GitHub source files).

The result is that:
- In *Related Pages* there is a *LICENSE* link that works (links to `docs-out/html/md_LICENSE.html`).
- In *Main Page* there is *License* link that does not work (links to `docs-out/html/LICENSE`).

**Comment by albert-github** [comment]
The file 'LICENSE' is not sen as a file to which a (markdown) link is possible. Best is to use:
```
HTML_EXTRA_FILES       = LICENSE
```

**Comment by qeef** [comment]
So I changed the config of `Doxyfile`:
```
EXTENSION_MAPPING      =
HTML_EXTRA_FILES       = LICENSE
```

It works for me. I am not sure if this issue should be still considered as a bug, though. IMHO it depends on what *extension mapping* means for Doxygen. If *just parse* as mapped format or *behave* as the mapped format.

From my point of view, the issue I had is solved. However, I can't decide if the issue should stay open.

**Comment by albert-github** [comment]
I think that both type of usage are valid usage so I think it is a (small) bug in doxygen.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7121

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
